waflib/Task.py

```diff
@@ -411,10 +411,11 @@ def format_error(self):
 		elif not self.hasrun:
 			return 'task in %r was not executed for some reason: %r' % (name, self)
 		elif self.hasrun == CRASHED:
+			msg = ' '.join(msg)
```

Comment:
I think it is possible that `msg` is also a string when the task's command is executed in "shell" mode. In that case, the msg would become a bunch of characters separated with spaces.